### PR TITLE
Fix multiline text property value renderer

### PR DESCRIPTION
**Pull request by simasjar**
## Changes
Closes ITwin/presentation#144

image

image

image

## Testing

N/A

### CODE REVIEW

File: ui/components-react/src/components-react/properties/renderers/value/MultilineTextPropertyValueRenderer.tsx [hunk 1]
```diff
@@ -96,21 +96,21 @@ export const MultilineTextRenderer: React.FC<MultilineTextRenderer> = (props) =>
     <div ref={divRef} className="multiline">
       <span
         ref={spanRef}
-        className={(numRerenders !== 0 && props.isExpanded) ? "expanded" : ""}
+        className={`multiline-property-span ${props.isExpanded ? "expanded" : ""}`}
```
Comment (saskliutas) [inline ui/components-react/src/components-react/properties/renderers/value/MultilineTextPropertyValueRenderer.tsx:99]:
```suggestion
        className={classnames("content", { expanded: props.isExpanded })}
```

File: ui/components-react/src/components-react/properties/renderers/value/MultilineTextPropertyValueRenderer.tsx [hunk 2]
```diff
@@ -64,19 +62,21 @@ interface MultilineTextRenderer {
 
 /** @internal */
 export const MultilineTextRenderer: React.FC<MultilineTextRenderer> = (props) => { // eslint-disable-line @typescript-eslint/no-redeclare
-  const { rerenderContext, numRerenders, rerender } = useRerender({ contentOverflows: false });
   const spanRef = useRef<HTMLSpanElement>(null);
   const divRef = useRef<HTMLDivElement>(null);
   const previousHeightRef = useRef(0);
+  const contentOverflows = useRef<boolean>(false);
+  const [numRerenders, setNumRerenders] = React.useState<number>(0);
```
Comment (saskliutas) [inline ui/components-react/src/components-react/properties/renderers/value/MultilineTextPropertyValueRenderer.tsx:69]: Is there a need to count numRerenders? I think it would be enough to put `contentOverflows` in state instead of ref.

**Review by saskliutas — CHANGES_REQUESTED** [review]

File: ui/components-react/src/components-react/properties/renderers/value/MultilineTextPropertyValueRenderer.tsx (hunk 2 above)
Comment (simasjar) [reply ui/components-react/src/components-react/properties/renderers/value/MultilineTextPropertyValueRenderer.tsx]: fixed

File: common/changes/@itwin/components-react/Fix_multiline_text_property_value_renderer_2023-05-08-12-34.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/components-react",
+      "comment": "Fix bug where MultilineTextPropertyValueRenderer doesn't show \"See less\" and \"See more\" buttons",
```
Comment (grigasp) [inline common/changes/@itwin/components-react/Fix_multiline_text_property_value_renderer_2023-05-08-12-34.json:5]:
```suggestion
      "comment": "Fix a bug where `MultilineTextPropertyValueRenderer` doesn't show \"See less\" and \"See more\" buttons",
```

**Review by grigasp — CHANGES_REQUESTED** [review]

File: ui/components-react/src/components-react/properties/renderers/PropertyView.scss [hunk 4]
```diff
@@ -73,10 +73,15 @@ $text-font-color: var(--iui-color-text);
...
+
+    .content {
+      overflow: hidden;
+      text-overflow: ellipsis;
+    }
   }
 
   .expanded {
-    white-space: pre-wrap;
+    white-space: pre;
```
Comment (saskliutas) [inline ui/components-react/src/components-react/properties/renderers/PropertyView.scss:84]: Why was this changed to `pre`? How does it work with long string that has spaces but no new line characters?

Comment (simasjar) [reply ui/components-react/src/components-react/properties/renderers/PropertyView.scss]: fixed

File: ui/components-react/src/components-react/properties/renderers/value/MultilineTextPropertyValueRenderer.tsx [hunk 5]
```diff
@@ -64,24 +64,19 @@ interface MultilineTextRenderer {
 
 /** @internal */
 export const MultilineTextRenderer: React.FC<MultilineTextRenderer> = (props) => { // eslint-disable-line @typescript-eslint/no-redeclare
-  const { rerenderContext, numRerenders, rerender } = useRerender({ contentOverflows: false });
   const spanRef = useRef<HTMLSpanElement>(null);
   const divRef = useRef<HTMLDivElement>(null);
   const previousHeightRef = useRef(0);
-  useLayoutEffect(() => {
+  const [contentOverflows, setContentOverflows] = useState<boolean>(false);
+  useLayoutEffect(() => { // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (saskliutas) [inline ui/components-react/src/components-react/properties/renderers/value/MultilineTextPropertyValueRenderer.tsx:71]: Why was it working before without disabling this lint rule?
